icu4c/source/common/mlbe.cpp

```diff
@@ -67,14 +67,14 @@ int32_t MlBreakEngine::divideUpRange(UText *inText, int32_t rangeStart, int32_t
     // Add a break for the start.
     boundary.addElement(0, status);
     numBreaks++;
-    if (U_FAILURE(status)) return 0;
-
-    for (int32_t idx = 0; idx + 1 < codePointLength && U_SUCCESS(status); idx++) {
-        numBreaks =
-            evaluateBreakpoint(inString, indexList, idx, numCodeUnits, numBreaks, boundary, status);
-        if (idx + 4 < codePointLength) {
-            indexList[idx + 6] = numCodeUnits;
-            numCodeUnits += U16_LENGTH(inString.char32At(indexList[idx + 6]));
+    if (U_SUCCESS(status)) {
+        for (int32_t idx = 0; idx + 1 < codePointLength && U_SUCCESS(status); idx++) {
+            numBreaks =
+                evaluateBreakpoint(inString, indexList, idx, numCodeUnits, numBreaks, boundary, status);
+            if (idx + 4 < codePointLength) {
+                indexList[idx + 6] = numCodeUnits;
+                numCodeUnits += U16_LENGTH(inString.char32At(indexList[idx + 6]));
+            }
         }
     }
     uprv_free(indexList);
```

Comment:
You could change indexList to be a LocalMemory<int32_t>. Get rid of the error prone naked pointer and explicit free() altogether.

Reply:
PTAL